torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3327,6 +3327,44 @@ std::tuple<Tensor, Tensor> polar_backward(
   return std::make_tuple(grad_abs, grad_angle);
 }
 
+Tensor i1_backward(
+    const Tensor& grad,
+    const Tensor& self,
+    const Tensor& result) {
+  return AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "i1_backward", [&]() {
+    // For x = 0, the correct gradient is 0.5,
+    // however due to floating point computation we get NaN.
+    // So we manually update gradient for x=0
+    auto eps = std::numeric_limits<scalar_t>::epsilon();
+    auto self_is_not_tiny = self.abs() > eps;
+    auto safe_self =
```

Comment:
Why do we need to perform the `where` here? I understand the comment but it seems to relate to the later `where` statement

Reply:
Related to this: https://github.com/pytorch/pytorch/issues/52248

where computes the gradient even when that value wasn't actually used. So without replacing the zeroes with `eps` leads to `nan` in the gradient computation.

Reply:
Oh; that's frustrating. Would you add a comment to that effect in the code so we can update this when that bug is fixed?